packages/react/src/transactions/publications/useCreatePostRequest.ts

```diff
@@ -16,8 +18,16 @@ function extractUnknownOpenActionModuleAddresses(desired: CreatePostRequest): Ev
 
 export function useCreatePostRequest() {
   const { execute: fetch } = useLazyModuleMetadata();
+  const { config } = useSharedDependencies();
+
   return async (input: UnknownObject): Promise<CreatePostRequest> => {
     const desired = createPostRequest(input);
+
+    // if sponsored is disabled globally, return here
+    if (config.sponsored === false) {
+      return overwriteSponsored(desired, config);
```

Comment:
I would make the decision explicit here, not sure if `overwriteSponsored` level of indirection is needed.